public/pages/utils/anomalyResultUtils.ts

```diff
@@ -84,21 +89,57 @@ export const getAnomalyResultsWithDateRange = (
       fieldName: AD_DOC_FIELDS.DATA_START_TIME,
     },
   };
-  dispatch(getDetectorResults(detectorId, updatedParams));
+};
+
+export const getAnomalyResultsWithDateRange = (
+  dispatch: Dispatch<any>,
+  startTime: number,
+  endTime: number,
+  detectorId: string
+) => {
+  const params = buildParamsForGetAnomalyResultsWithDateRange(
+    startTime,
+    endTime
+  );
+  dispatch(getDetectorResults(detectorId, params));
 };
 
 const MAX_DATA_POINTS = 1000;
+const MAX_FEATURE_ANNOTATIONS = 100;
 
 const calculateStep = (total: number): number => {
   return Math.ceil(total / MAX_DATA_POINTS);
 };
 
-// If array size is 100K, `findAnomalyWithMaxAnomalyGrade` 
-// takes less than 2ms by average, while `Array#reduce` 
-// takes about 16ms by average and`Array#sort` 
+const calculateSampleWindowsWithMaxDataPoints = (
```

Comment:
Feature data points are already sampled, should we sample again for the missing data points? Can we show missing alert annotation for every feature data points.

Reply:
Using sampled feature data points in will cause that some missing data points can not be captured, because when we sample anomaly results, we collect all the data for fixed [time range](https://github.com/opendistro-for-elasticsearch/anomaly-detection-kibana-plugin/blob/master/public/pages/utils/anomalyResultUtils.ts#L314-L316), if both missing data points and existing data points are there in that range, we will not be able to catch the missing ones.